.github/workflows/ci.yaml

```diff
@@ -144,32 +158,26 @@ jobs:
 
   # Separate job to pre-populate the base dependency cache
   # This prevent upcoming jobs to do the same individually
-  prepare-base:
-    name: Prepare base dependencies
```

Comment:
Tbh I did appreciate having two separate `prepare` tasks with `prepare-base` and `prepare-tests`. Usually `prepare-tests` does take quite a bit longer, so all jobs which didn't depend on it could start early and be done when pytest is run. It's just a nice experience to get most linter results pretty fast. Especially when requirements change, `prepare-base` still got a cache hit.

Reply:
It's an extra job that takes up time and queuing. When changing deps, it will still run the full suite anyways. That minute won't make a difference.

Reply:
I think it might maybe make more sense to split out the pre-commit cache creation, which will allow setting of most linters almost instantly (even faster compared to the previous situation).